src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/Payments/RefreshPaymentData/RefreshPaymentDataCommandHandler.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EAS.Application.Events;
+using SFA.DAS.EAS.Application.Events.ProcessPayment;
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Data;
+using SFA.DAS.Payments.Events.Api.Client;
+
+namespace SFA.DAS.EAS.Application.Commands.Payments.RefreshPaymentData
+{
+    public class RefreshPaymentDataCommandHandler : AsyncRequestHandler<RefreshPaymentDataCommand>
+    {
+        private readonly IValidator<RefreshPaymentDataCommand> _validator;
+        private readonly IPaymentsEventsApiClient _paymentsEventsApiClient;
+        private readonly IDasLevyRepository _dasLevyRepository;
+        private readonly IMediator _mediator;
+
+        public RefreshPaymentDataCommandHandler(IValidator<RefreshPaymentDataCommand> validator, IPaymentsEventsApiClient paymentsEventsApiClient, IDasLevyRepository dasLevyRepository, IMediator mediator)
+        {
+            _validator = validator;
+            _paymentsEventsApiClient = paymentsEventsApiClient;
+            _dasLevyRepository = dasLevyRepository;
+            _mediator = mediator;
+        }
+
+        protected override async Task HandleCore(RefreshPaymentDataCommand message)
+        {
+            var validationResult = _validator.Validate(message);
+
+            if (!validationResult.IsValid())
+            {
+                throw new InvalidRequestException(validationResult.ValidationDictionary);
+            }
+
+            var payments = await _paymentsEventsApiClient.GetPayments(message.PeriodEnd, message.AccountId.ToString());
+            var sendPaymentDataChanged = false;
+
+            foreach (var payment in payments.Items)
+            {
+                var existingPayment = await _dasLevyRepository.GetPaymentData(Guid.Parse(payment.Id));
```

Comment:
Agree, for now we just need to add this as a tech task given it doesn't just affect this story. I'll add a tech task now to review this sort of issue system wide.